adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -556,9 +587,15 @@ private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device a
 
         if (!isPayloadOfIndicatedType(payload, ctx.contentType())) {
             return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
-                    String.format("Content-Type %s does not match with the payload", ctx.contentType())));
+                    String.format("Content-Type %s does not match with payload", ctx.contentType())));
         } else {
 
+            Optional.ofNullable(getCurrentSpan(ctx)).map(span -> {
+                span.setOperationName("upload " + endpointName);
+                TracingHelper.TAG_AUTHENTICATED.set(span, ctx.authenticatedDevice() != null);
+                return span.context();
+            }).orElse(null);
```

Comment:
I would have expected an assignment to a `SpanContext` variable here - otherwise the `.orElse(null)` at the end is superfluous. And in that case a simple `if (span != null)` block would look more readable imho (also as `return span.context()` is ignored).